primitives/core/src/bounded/bounded_vec.rs

```diff
@@ -720,6 +738,42 @@ impl<T, S: Get<u32>> TruncateFrom<Vec<T>> for BoundedVec<T, S> {
 	}
 }
 
+impl<T, S: std::cmp::PartialOrd<T> + std::clone::Clone> Min<T> for S {
+	fn min(&self, t: T) -> Result<Self, T> {
+		if self <= &t {
+			Ok(self.clone())
+		} else {
+			Err(t)
+		}
+	}
+
+	fn strict_min(&self, t: T) -> Result<Self, T> {
+		if self < &t {
+			Ok(self.clone())
+		} else {
+			Err(t)
+		}
+	}
+}
+
+impl<T, S: std::cmp::PartialOrd<T> + std::clone::Clone> Max<T> for S {
```

Comment:
Looks better now. Not what the best style here is.  
Do we want to add Min/Max traits? Or just the functions without trait?

Reply:
I think with the traits. Since that's the kind of implementation that's suggested [here](https://github.com/paritytech/substrate/pull/12554#discussion_r1004158204)

Reply:
Oliver's suggestion is better -- we can directly use `PartialOrd` and such in the blanket impl of `trait DefenisveMin`, no need for two traits.

Reply:
I have gone ahead to do as such @kianenigma @ggwpez